samples/contrib/versioned-pipeline-ci-samples/kaggle-ci-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,182 @@
+        "build",
+        "-t",
+        "${_GCR_PATH}/kaggle_download:$COMMIT_SHA",
+        "-t",
+        "${_GCR_PATH}/kaggle_download:latest",
+        "${_CODE_PATH}/download_dataset",
+        "-f",
+        "${_CODE_PATH}/download_dataset/Dockerfile",
+      ]
+    id: "BuildDownloadDataImage"
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "push",
+        "${_GCR_PATH}/kaggle_download:$COMMIT_SHA",
+      ]
+    id: "PushDownloadDataImage"
+    waitFor: ["BuildDownloadDataImage"]
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "build",
+        "-t",
+        "${_GCR_PATH}/kaggle_visualize_table:$COMMIT_SHA",
+        "-t",
+        "${_GCR_PATH}/kaggle_visualize_table:latest",
+        "${_CODE_PATH}/visualize_table",
+        "-f",
+        "${_CODE_PATH}/visualize_table/Dockerfile",
+      ]
+    id: "BuildVisualizeTableImage"
+  
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "push",
+        "${_GCR_PATH}/kaggle_visualize_table:$COMMIT_SHA",
+      ]
+    id: "PushVisualizeTableImage"
+    waitFor: ["BuildVisualizeTableImage"]
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "build",
+        "-t",
+        "${_GCR_PATH}/kaggle_visualize_html:$COMMIT_SHA",
+        "-t",
+        "${_GCR_PATH}/kaggle_visualize_html:latest",
+        "${_CODE_PATH}/visualize_html",
+        "-f",
+        "${_CODE_PATH}/visualize_html/Dockerfile",
+      ]
+    id: "BuildVisualizeHTMLImage"
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "push",
+        "${_GCR_PATH}/kaggle_visualize_html:$COMMIT_SHA",
+      ]
+    id: "PushVisualizeHTMLImage"
+    waitFor: ["BuildVisualizeHTMLImage"]
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "build",
+        "-t",
+        "${_GCR_PATH}/kaggle_train:$COMMIT_SHA",
+        "-t",
+        "${_GCR_PATH}/kaggle_train:latest",
+        "${_CODE_PATH}/train_model",
+        "-f",
+        "${_CODE_PATH}/train_model/Dockerfile",
+      ]
+    id: "BuildTrainImage"
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "push",
+        "${_GCR_PATH}/kaggle_train:$COMMIT_SHA",
+      ]
+    id: "PushTrainImage"
+    waitFor: ["BuildTrainImage"]
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "build",
+        "-t",
+        "${_GCR_PATH}/kaggle_submit:$COMMIT_SHA",
+        "-t",
+        "${_GCR_PATH}/kaggle_submit:latest",
+        "${_CODE_PATH}/submit_result",
+        "-f",
+        "${_CODE_PATH}/submit_result/Dockerfile",
+      ]
+    id: "BuildSubmitImage"
+
+  - name: "gcr.io/cloud-builders/docker"
+    args:
+      [
+        "push",
+        "${_GCR_PATH}/kaggle_submit:$COMMIT_SHA",
+      ]
+    id: "PushSubmitImage"
+    waitFor: ["BuildSubmitImage"]
+
+  - name: "python:3.7-slim"
+    entrypoint: "/bin/sh"
+    args: [
+        "-c",
+        "cd ${_CODE_PATH};
+        pip3 install cffi==1.12.3 --upgrade;
```

Comment:
Do we wish to 'set -ex' here for 1) perhaps better logging, and 2) operation atomicity?

Reply:
Thank you for the advice